src/ch03-02-01-madara.md

```diff
@@ -1,11 +1,420 @@
 # Madara 🚧
 
-> **TODO:** ADD EXAMPLES OF HOW TO SET UP AND USE MADARA
-
 Madara is a Starknet sequencer that operates on the Substrate framework, executing Cairo programs and Starknet smart contracts with the Cairo VM. Madara enables the launch and control of Starknet Appchains or L3s.
 
 ## Get Started with Madara
 
-Visit the [GitHub repository](https://github.com/keep-starknet-strange/madara) for detailed instructions on installing and configuring Madara, including practical examples.
+<!-- Visit the [GitHub repository](https://github.com/keep-starknet-strange/madara) for detailed instructions on installing and configuring Madara, including practical examples.
+
+
+
+# Building on Madara -->
+
+<br></br>In this section, we will guide you through the building process so you can start hacking on the Madara stack.
+We will go from running your chain locally to changing the consensus algorithm and interacting with smart contracts on your own chain!
+
+## Let's start
+
+<Steps>
+
+### Install dependencies
+
+<a id="dependencies"></a>
+We first need to make sure you have everything needed to complete this tutorial.
+
+| Dependency | Version              | Installation                                                                                                  |
+| ---------- | -------------------- | ------------------------------------------------------------------------------------------------------------- |
+| Rust       | rustc 1.69.0-nightly | `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs \| sh` <br></br>`rustup toolchain install nightly` |
+| nvm        | latest               | `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh \| bash`                            |
+| Cairo      | 1.0                  | `curl -L https://github.com/franalgaba/cairo-installer/raw/main/bin/cairo-installer \| bash`                  |
+
+for macos ensure you have protobuf to avoid build time errors
+
+```
+brew install protobuf
+```
+
+### Build the chain
+
+We will spin up a CairoVM Rollup from the Madara Stack source code.
+You could use docker images, but this way we keep the option to modify component behavior if you need to do so.
+The Madara stack source code is a monorepo which can be found [here](https://github.com/keep-starknet-strange/madara)
+
+```bash
+cd ~
+git clone https://github.com/keep-starknet-strange/madara.git
+```
+
+Then let's build the chain in `release` mode
+
+```bash
+cd madara
+cargo build --release
+```
+
+<br></br>
+
+**Single-Node Development Chain**
+
+This command will start the single-node development chain with non-persistent
+state:
+
+```bash
+./target/release/madara --dev
```

Comment:
![Screenshot from 2024-02-12 12-03-39](https://github.com/starknet-edu/starknetbook/assets/22782504/3e87bbd4-2554-4bf1-a8b2-91a772081d71)

I got this error at this point

![Screenshot from 2024-02-12 12-05-29](https://github.com/starknet-edu/starknetbook/assets/22782504/aa128194-e087-4d96-9d9f-0c5c69583d57)

There is not regenisis file? Do you have any idea how to solve it?